frontend/public/components/data-view-poc/DataViewPodList.tsx

```diff
@@ -0,0 +1,801 @@
+}
+
+// Mostly copied over from public/components/filter-toolbar.tsx
+// Only taken the labels filter as it is unique to OCP and not in data view
+// Seems like it could be generalized and transferred as a new filter type to data view
+const LabelsFilter = ({
+  onFilterChange,
+  data,
+  title,
+  filterId,
+  showToolbarItem,
+}: {
+  onFilterChange: OnFilterChange;
+  data: PodKind[];
+  filterId: string;
+  title: string;
+  showToolbarItem?: boolean;
+}) => {
+  const { t } = useTranslation();
+  const [labelInputText, setLabelInputText] = React.useState('');
+  const [searchParams] = useSearchParams();
+
+  const [labelSelection, onLabelSelectionChange] = useLabelSelectionFix(searchParams, filterId);
+  const applyLabelFilters = (values: string[]) => {
+    setLabelInputText('');
+    onLabelSelectionChange(values);
+    onFilterChange(filterId, { all: values });
+  };
+
+  return (
+    <ToolbarFilter
+      deleteLabelGroup={() => {
+        setLabelInputText('');
+        applyLabelFilters([]);
+      }}
+      labels={labelSelection}
+      deleteLabel={(f, chip: string) => {
+        setLabelInputText('');
+        applyLabelFilters(_.difference(labelSelection, [chip]));
+      }}
+      categoryName={title}
+      showToolbarItem={showToolbarItem}
+    >
+      <div className="pf-v6-c-input-group co-filter-group">
+        <AutocompleteInput
+          className="co-text-node"
+          onSuggestionSelect={(selected) => {
+            applyLabelFilters(_.uniq([...labelSelection, selected]));
+          }}
+          showSuggestions
+          textValue={labelInputText}
+          setTextValue={setLabelInputText}
+          placeholder={t('public~Search by label...')}
+          data={data}
+        />
+      </div>
+    </ToolbarFilter>
+  );
+};
+
+type DataViewPodListProps = {
+  loaded: boolean;
+  data: PodKind[];
+  showNodes?: boolean;
+  filters?: RowFilter<PodKind>[];
+  onFilterChange: OnFilterChange;
+  columnLayout?: ColumnLayout;
+  showNamespaceOverride?: boolean;
+};
+
+const DataViewPodList = ({
+  data,
+  showNodes,
+  loaded,
+  filters,
+  onFilterChange,
+  columnLayout,
+  showNamespaceOverride,
+}: DataViewPodListProps) => {
+  const { t } = useTranslation();
+  const { dataViewColumns, dataViewRows, pagination } = useDataViewData({
+    data,
+    showNodes,
+    showNamespaceOverride,
+  });
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialFilters = filters?.reduce((acc, curr) => {
+    acc[curr.filterGroupName] = searchParams.getAll(curr.filterGroupName);
+    return acc;
+  }, {});
+  const debouncedOnFilterChange = useDebounceCallback(onFilterChange, 500);
+  const debouncedSetSearchParams = useDebounceCallback(setSearchParams, 500);
+  // Currently there is a big that will only return the first item from the query, even though there are multiple items for one group
+  const { onSetFilters, filters: dataViewFilters } = useDataViewFilters({
```

Comment:
`clearAllFilters` is missing here.  I added it and passed it to `<DataViewToolbar>`, but it's not working.  Any ideas?

Reply:
Thinking this is a PatternFly bug.

Reply:
OK I don't think its something specific to data view, I think it might be something with how I integrated it into the OCP. I'll poke around for a bit now.

Reply:
@rhamilto ok I've added the callback here: https://github.com/openshift/console/pull/14897/commits/dd3ecf8c44135a53d73d59dea774ffc80b7bbad3

its almost working, I think with how the filtering is wired in this component in general it will need some extra bits than just using the standard callback.

Reply:
Ok.  I cobbled together [a POC that only uses data view filters](https://github.com/openshift/console/pull/15057/commits) and everything works as expected.